src/synapse/azext_synapse/_params.py

```diff
@@ -0,0 +1,96 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+# pylint: disable=line-too-long
+
+from azure.cli.core.commands.parameters import name_type, tags_type
+
+
+# pylint: disable=too-many-statements
+def load_arguments(self, _):
+    # synapse spark
+    with self.argument_context('synapse spark') as c:

```

Comment:
This adds all below arguments for all commands starting with `synapse spark` including list and get.

Reply:
> This adds all below arguments for all commands starting with `synapse spark` including list and get.

I have tested. They won't because there are specific argument context such as ‘ synapse spark batch’ and 'synapse spark batch'.
Because there are some common argument for ‘ synapse spark batch’ and 'synapse spark batch'. So I add a 'synapse spark' in order reuse the argument.

Reply:
> > This adds all below arguments for all commands starting with `synapse spark` including list and get.
> 
> I have tested. They won't because there are specific argument contexts such as ‘ synapse spark batch’ and 'synapse spark session'.
> Because there are some common argument for ‘ synapse spark batch’ and 'synapse spark batch'. So I add a 'synapse spark' in order reuse the argument.

Reply:
I think the arguments are not showed in help for list/show because the code first check the arguments in the method in custom.py and then look for the help text here. So this also works because the arguments in custom.py is correct. 

But it's a little bit confusing because they are only common arguments for create/update, not for all commands starting with `synapse spark`. You can use scope  to share common arugments for synapse spark batch create/update, synapse spark session create/update. See some examples in current code:[1](https://github.com/Azure/azure-cli/blob/dev/src/azure-cli/azure/cli/command_modules/appservice/_params.py#L165), [2](https://github.com/Azure/azure-cli/blob/dev/src/azure-cli/azure/cli/command_modules/eventhubs/_params.py#L56..L67).

Reply:
It seems only `workspace_name` and `spark_pool_name` are the common arguments for all commands starting with `synapse spark`. You can leave them here.